stubs/urllib3/urllib3/contrib/socks.pyi

```diff
@@ -0,0 +1,41 @@
+from collections.abc import Mapping
+from typing_extensions import TypedDict
+from typing import Any, ClassVar
+
+from ..connection import HTTPConnection, HTTPSConnection
+from ..connectionpool import HTTPConnectionPool, HTTPSConnectionPool
+from ..poolmanager import PoolManager
+
+class _SockOptions(TypedDict):
+    socks_version: int
+    proxy_host: str | None
+    proxy_port: str | None
+    username: str | None
+    password: str | None
+    rdns: bool
+
+class SOCKSConnection(HTTPConnection):
+    def __init__(self, _socks_options: _SockOptions, *args: Any, **kwargs: Any) -> None: ...
```

Comment:
As we might reasonably be able to type `args` and `kwargs`:

```suggestion
    def __init__(self, _socks_options: _SockOptions, *args, **kwargs) -> None: ...
```

Reply:
Updated. How would I type hint these given that they are `args` / `kwargs` of the `HTTPConnection` as seen in this?

https://github.com/urllib3/urllib3/blob/1.26.x/src/urllib3/connection.py#L115

With type hints in 2.x branch (missing the strict parameter):
https://github.com/urllib3/urllib3/blob/main/src/urllib3/connection.py#L136-L148

Reply:
We would basically copy the signature of `HTTPConnection.__init__()`. Cumbersome, but possible.